core/src/main/java/org/mskcc/cbio/oncokb/util/SummaryUtils.java

```diff
@@ -254,6 +254,22 @@ private static String getOncogenicSummarySubFunc(Gene gene, Alteration exactMatc
             queryAlteration = queryAlteration.substring(0, 1).toUpperCase() + queryAlteration.substring(1);
         }
 
+        // if the gene is Other Biomarker, return the mutation effect description for alteration instead
+        if (gene.getHugoSymbol().equals(SpecialStrings.OTHERBIOMARKERS)) {
+            if (exactMatchAlteration != null) {
+                List<Evidence> evidences = EvidenceUtils.getEvidence(Collections.singletonList(exactMatchAlteration), Collections.singleton(EvidenceType.MUTATION_EFFECT), null);
+
+                // Technically the list should only contain no more than one record.
+                if (evidences.size() > 0) {
+                    return evidences.iterator().next().getDescription();
```

Comment:
Should you use while and iterator. hasNext() to loop through the list?
From what you currently written, it simply return the description of first element in evidences, if I understood things correctly.